core/alpaca/alpaca_trade_manager.py

```diff
@@ -47,16 +47,19 @@ def buy_stock(self, ticker: str) -> None:
         purchase_amnt = round(buying_power * 0.05, 2)
         floor = round(purchase_amnt * 0.9, 2)
 
-        self.api.submit_order(
-            symbol=ticker,
-            notional=purchase_amnt,
-            stop_loss=dict(
-              stop_price=floor
-            ),
-            side='buy',
-            type='market',
-            time_in_force='day'
-        )
+        try:
+            self.api.submit_order(
+                symbol=ticker,
+                notional=purchase_amnt,
+                stop_loss=dict(
+                stop_price=floor
+                ),
+                side='buy',
+                type='market',
+                time_in_force='day'
+            )
+        except:
```

Comment:
It would be best to catch the specific exception here.
```
except APIError as e:
    logging.error(f"APIError occurred: {e}")

_Note you'll have to import APIError from alpaca_trade_api.rest_
```

Reply:
ok I tried doing that without an import statement

Reply:
I tried to get the API Error from AlpacaTradeManager.rest but I guess I confused the objects